File: 1089-duplicate-zeros/1089-duplicate-zeros.py
class Solution(object):
    def duplicateZeros(self, arr):
        """
        :type arr: List[int]
        :rtype: None Do not return anything, modify arr in-place instead.
        """
        # Store old length
        oldLen = len(arr) 
        i = 0
        
        while i < oldLen :
            # Insert 0 into list
            if arr[i] == 0 :
                arr.insert(i+1 , 0)
                i += 1
            i += 1
        # Snip arr to old length
        arr[:] = arr[:oldLen]

File: 1089-duplicate-zeros/1089-duplicate-zeros.js
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
var duplicateZeros = function(arr) {
    // Iterate through list
    for (var i = 0; i < arr.length; i++) {
        // Check if that index has a zero
        if (arr[i] === 0) {
            // Splice and Delete if there is a trailing zero
            arr.splice(i, 0, 0);
            arr.pop();
            i += 1
        }
    }
};